frontend/src/components/FilterSection.tsx

```diff
@@ -0,0 +1,58 @@
+import "./FilterSection.css";
+import { Badge, Card } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import { useSearchParams } from "../hooks/useSearchParams.ts";
+import { X } from "react-bootstrap-icons";
+
+export const FilterSection = ({ languages }: { languages: string[] }) => {
+  const params = useSearchParams();
+
+  return (
+    <Card className="mx-auto mb-3 w-75">
+      <Card.Header>Filtering</Card.Header>
+      <Card.Body>
+        <Card.Title>Language</Card.Title>
+        <div className="d-flex flex-wrap gap-2 py-2">
+          {languages.map((lang) => (
+            <FilterChip key={lang} current={lang} searchParams={params} />
+          ))}
+        </div>
+      </Card.Body>
+    </Card>
+  );
+};
+
+const FilterChip = ({
+  current,
+  searchParams,
+}: {
+  current: string;
+  searchParams: URLSearchParams;
+}) => {
+  const languages = searchParams.getAll("language");
+  if (languages.includes(current)) {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.delete("language");
+    languages
+      .filter((lang) => lang !== current)
+      .forEach((lang) => newParams.append("language", lang));
+    return (
+      <Link key={current} to={{ search: newParams.toString() }}>
+        <Badge pill bg="filter-active">
+          <div className="px-2">{current}</div>
```

Comment:
Slightly nice to titlecase the names here so we see `Python` rather than `python` -- right now in particular for some reason `Kotlin` is the only language title-ized.

Even nicer would probably be mapping these names to arbitrary human readable names -- in particular probably it's nicer to see `.NET` rather tha `Dotnet`.

But OK with at least the former (title casing).

Reply:
> Languages returned from harnesses should be somehow standardized so there is no such case that one implementation is in python and other one in Python.

This is regardless indeed a good point, so fixing that (to validate that the returned value is lowercase, and then fixing the Kotlin harness to return it that way)

Reply:
(Done)